pkg/devfile/adapters/docker/component/utils.go

```diff
@@ -25,36 +25,17 @@ import (
 )
 
 const (
-	localhostIP             = "127.0.0.1"
-	projectSourceVolumeName = "odo-project-source"
+	localhostIP = "127.0.0.1"
 )
 
 func (a Adapter) createComponent() (err error) {
 	componentName := a.ComponentName
 
 	log.Infof("\nCreating Docker resources for component %s", a.ComponentName)
 
-	// Get or create the project source volume
-	var projectVolumeName string
-	projectVolumeLabels := utils.GetProjectVolumeLabels(componentName)
-	projectVols, err := a.Client.GetVolumesByLabel(projectVolumeLabels)
+	projectVolumeName, err := utils.CreateAndGetProjectVolume(a.Client, componentName)
```

Comment:
I'm wondering (and want your thoughts on this), if `CreateAndGetProjectVolume` returns an already existing volume (potentially with old/outdated/incomplete source code on it) during first-time push, should it overwrite the contents on the volume?

Reply:
so if its a first time push, i.e.; `componentExists` is false, we can force sync if a volume already exists. Yeah i like the idea, it sounds like we should do it

Reply:
it looks like our current master code already covers this.

If its first time push, componentExists = false, we sync. Doesn't matter if we are reusing volume.

if its update component, componentExists = true, we sync 
- if any of the container from devfile component is missing
- if any of the container needs to be updated
- if containers does not require update, files changed, indexer handles it.

So looks like i might not need to do anything.